fastlane/Fastfile

```diff
@@ -248,6 +256,74 @@ lane :download_translations do |commit_changes: false|
   end
 end
 
+# Synchronizes localization files with GlotPress and creates a PR
+#
+# This lane performs a complete localization sync:
+# 1. Generates source PO file from English Fluent file
+# 2. Downloads latest translations from GlotPress
+# 3. Creates a PR with the changes instead of pushing directly to trunk
+#
+lane :sync_localization do
+  github_token = ENV['GITHUB_TOKEN'] || UI.user_error!('GITHUB_TOKEN environment variable is required')
+
+  sync_branch = 'localization/sync-translations'
+
+  Fastlane::Helper::GitHelper.delete_local_branch_if_exists!(sync_branch)
+  Fastlane::Helper::GitHelper.create_branch(sync_branch)
+
+  po_result = generate_source_po_file(commit_changes: true, skip_date_only_changes: true)
+  translations_result = download_translations(commit_changes: true)
+
+  # Check if either operation resulted in changes
+  has_changes = !po_result.nil? || !translations_result.nil?
+  unless has_changes
+    UI.important('⚠️ No changes to commit')
+
+    if is_ci
+      buildkite_annotate(
```

Comment:
Not for this PR, but we should probably include that `if is_ci` check as part of the `def self.run` implementation of our `buildkite_*` actions from `release-toolkit` at one point…